### PR TITLE
Upgrade jackc/pgx to v5

### PR DESCRIPTION
**Pull request by alnr**
### What is being done in this PR?

Closes https://github.com/gobuffalo/pop/issues/818

Supersedes https://github.com/gobuffalo/pop/pull/814

### What are the main choices made to get to this solution?
jackc/pgx/v5 has slightly different handling of SQL NULLs than v4. Otherwise, this was a drop-in replacement.

### List the manual test cases you've covered before sending this PR:
Ran tests against postgres locally.

### CODE REVIEW

File: slices/float.go [hunk 1]
```diff
@@ -20,6 +20,9 @@ func (f Float) Interface() interface{} {
 func (f *Float) Scan(src interface{}) error {
 	var str string
 	switch t := src.(type) {
+	case nil:
```
Comment (sio4) [inline slices/float.go:23]: Could it be changed to the general `nil` checking like the `String` case for readability? The other types too, including `Map`.

**Review by sio4 — CHANGES_REQUESTED** [review]
I left a comment on `Float` which could be applied to the others too. Otherwise, it looks good to me!
Thanks!

File: slices/float.go (hunk 1 above)
Comment (alnr) [reply slices/float.go]: Done.

Comment (alnr) [reply slices/float.go:23]: Actually, that change broke the whole thing haha. We must distinguish between a nil interface and a non-nil interface backed by nil.

I've changed the `String` case to match the other implementations.

**Comment by alnr** [comment]
pgx/v5 is a pretty big change from v4. I've changed this PR back to draft while waiting for some insights here: https://github.com/jackc/pgx/issues/1566

File: slices/float.go (hunk 1 above)
Comment (sio4) [reply slices/float.go:23]: maybe a different confusion here. if the type switch's `case nil` works like your PR, that means the `src` has no specific type and is actually `nil` so `src == nil` should be true. If the `src` is a "non-nil interface with a `nil` value", the type of `src` can't be `nil` and the `case nil` will not work. What is the actual `src` in here? something like `*[]string(nil)`?

I hope we have enough description on the PR or linked issue so we can have a solid reference for the chance in the code history.

**Comment by alnr** [comment]
Superseded by https://github.com/gobuffalo/pop/pull/835

**Pull request closed without merging**